session/pingpong/accountant_caller.go

```diff
@@ -206,4 +211,27 @@ var accountantCauseToError = map[string]error{
 	ErrAccountantNoPreviousPromise.Error():        ErrAccountantNoPreviousPromise,
 	ErrAccountantHashlockMissmatch.Error():        ErrAccountantHashlockMissmatch,
 	ErrAccountantNotFound.Error():                 ErrAccountantNotFound,
+	ErrNeedsRRecovery.Error():                     ErrNeedsRRecovery,
+}
+
+var encryptedRRegexp = regexp.MustCompile(`Encrypted recovery data: "(?P<recoverydetails>[0-9a-f]+)"`)
+
+type rRecoveryDetails struct {
+	R           string `json:"r"`
+	AgreementID uint64 `json:"agreement_id"`
+}
+
+func parseErrNeedsRRecoveryDetails(in string) (string, error) {
```

Comment:
Looks bogus. Why parsing from error message is needed at all?

Reply:
Because I don't want to do extra requests.

Reply:
I think what Waldz meant to ask is: perhaps it's possible to use a more convenient struct than a string?

Reply:
Agree. Maybe returning recovery details in separate field could help?